library/std/src/sys/thread_local/native/eager.rs

```diff
@@ -30,16 +30,22 @@ impl<T> Storage<T> {
     /// The `self` reference must remain valid until the TLS destructor is run.
     #[inline]
     pub unsafe fn get(&self) -> *const T {
-        match self.state.get() {
-            State::Alive => self.val.get(),
-            State::Destroyed => ptr::null(),
-            State::Initial => unsafe { self.initialize() },
+        if let State::Alive = self.state.get() {
+            self.val.get()
+        } else {
+            unsafe { self.get_or_init_slow() }
         }
     }
 
     #[cold]
-    unsafe fn initialize(&self) -> *const T {
```

Comment:
Should this be marked as `#[inline(never)]`? I've noticed `destructors::register` calls being inlined into the hot-path before.